pkg/generators/openapi_test.go

```diff
@@ -2386,7 +2387,7 @@ func TestMarkerComments(t *testing.T) {
 			// +k8s:validation:pattern="^foo$[0-9]+"
 			StringValue string
 
-			// +k8s:validation:maxitems=10
+			// +k8s:validation:maxItems=10
```

Comment:
Wondering why this passed the test before?

Reply:
Because maxItems is never used to check anything in the test :(  If the test had tried to exceed maxItems, it would have been allowed.

I could just remove this validation rules to simplify the test case..

Reply:
hmmm.. I think it is because the `json.Unmarshal` is [not case-sensitive](https://pkg.go.dev/encoding/json#Unmarshal). So when unmarshal the `commentTags` here https://github.com/kubernetes/kube-openapi/blob/99599406b04fd8d4b08b519eef9461744667043d/pkg/generators/markers.go#L411 , `maxItems` and `maxitems` both are parsed.

Reply:
We should not rely on that, I think the JSON we use in k/k is not case-insensitve

Reply:
k/k is using https://github.com/kubernetes-sigs/json which is case sensitive

Reply:
I'll follow up on this in a separate issue.